pkg/controller/cf_update_handlers.go

```diff
@@ -341,8 +343,11 @@ func (env *CfEnvironment) createAppServiceGraph(appId string, extIps []string) a
 			apicExtNet(name, cont.config.AciVrfTenant,
 				cont.config.AciL3Out, extIps))
 
-		serviceObjs = append(serviceObjs,
-			apicContract(name, cont.config.AciVrfTenant, graphName))
+		contract, err := apicContract(name, cont.config.AciVrfTenant, graphName, conScope)
+		if err != nil {
+			panic(err.Error())
```

Comment:
Let's replace the panic with an error return. Please also open an issue to track the need for validators for this annotation.